signals.go

```diff
@@ -0,0 +1,30 @@
+package main
```

Comment:
This signals stuff is pretty useful. I wonder if its worth moving into a package in a shared repo?